### PR TITLE
Replace old GitHub action with updated one.

### PR DESCRIPTION
**Pull request by uhafner**
The old one still used unsupported Java 11 JDK.

### CODE REVIEW

**Review by timja — APPROVED** [review]

**Comment by timja** [comment]
BuildStatusChecksPublisherITest.shouldPublishStageDetails:185 

fails on windows on Java 21

**Comment by uhafner** [comment]
@timja Do you have an idea why the output is `p1s2` without a time on Windows 21?
I even did not found the code that produces this part. 

```
Optional[  * Simple Stage *(15 ms)*
  * In parallel *(0.15 sec)*
    * p1 *(15 ms)*
      * p1s1 *(32 ms)*
        **Unstable**: *something went wrong*
      * p1s2
    * p2 *(15 ms)*
  * Fails *(31 ms)*
    **Error**: *a fatal error occurs*
```

**Comment by timja** [comment]
not sure =/

File: src/main/java/io/jenkins/plugins/checks/status/FlowExecutionAnalyzer.java [hunk 1]
```diff
@@ -89,7 +89,7 @@ private Pair<String, String> processStageOrBranchRow(final FlowGraphTable.Row ro
         if (row.getNode().isActive()) {
             nodeTextBuilder.append(" *(running)*");
         }
-        else if (row.getDurationMillis() > 0) {
+        else {
```
Comment (uhafner) [inline src/main/java/io/jenkins/plugins/checks/status/FlowExecutionAnalyzer.java:92]: @timja I removed the guard as `FlowGraphTable.Row` correctly handles the case where duration = 0. This makes the test run on all platforms now.

**Review by timja — APPROVED** [review]

**Pull request merged**